packages/minter-contracts/test-hs/Test/EnglishAuction/TezFixedFee.hs

```diff
@@ -0,0 +1,695 @@
+  , testExtendTime :: Natural
+  , testMaxConfigToStartTime :: Natural
+  , testTimeToStart :: Integer
+  , testAuctionDuration :: Natural
+  , testMaxAuctionTime :: Natural
+
+  , testOpeningPrice :: Mutez
+  , testFirstBid :: Mutez
+  , testMinRaise :: Mutez
+  , testMinRaisePercent :: Natural
+
+  , testTokenBatches :: [[FA2Token]]
+  }
+  deriving stock (Show)
+
+genTestData :: Gen TestData
+genTestData = do
+  -- TODO: fix bug
+  -- `feePercent` can technically be 0, but that would cause some properties to fail.
+  testFeePercent <- Gen.integral (Range.linear 1 100)
+  testRoundTime <- Gen.integral (Range.linear 1 100)
+  testExtendTime <- Gen.integral (Range.linear 1 100)
+
+  testMaxConfigToStartTime <- Gen.integral (Range.linear 0 100)
+  testTimeToStart <- Gen.integral (Range.linear 0 (toInteger testMaxConfigToStartTime))
+
+  -- TODO: fix bug
+  -- `openingPrice` can technically be in the [1, 100] range, but that could
+  -- sometimes cause the fee to be rounded down to 0, and therefore break some properties.
+  testOpeningPrice <- genMutez' (Range.linear 100 1000)
+  testFirstBid <- genMutez' (Range.linear testOpeningPrice (testOpeningPrice + 1000))
+  testMinRaise <- genMutez' (Range.linear 0 1000)
+  testMinRaisePercent <- Gen.integral (Range.linear 0 100)
+
+  testAuctionDuration <- Gen.integral (Range.linear 1 100)
+  testMaxAuctionTime <- Gen.integral (Range.linear testAuctionDuration (testAuctionDuration * 2))
+
+  testTokenBatches <-
+    Gen.list (Range.linear 0 10) $
+        Gen.list (Range.linear 0 10) $
+        FA2Token
+          <$> (TokenId <$> Gen.integral (Range.linear 0 10))
+          <*> Gen.integral (Range.linear 0 10)
+
+  pure $ TestData {..}
+
+-- | Generates a list with 0 or more valid bids
+genSomeBids :: TestData -> Gen (NonEmpty Mutez)
+genSomeBids testData = do
+  len <- Gen.int (Range.linear 0 5)
+  firstBid <- genBid testData (testOpeningPrice testData)
+  moreBids <- iterateM len (genBid testData) firstBid
+  pure $ firstBid :| moreBids
+
+-- | Generates a list with 1 or more valid bids
+genManyBids :: TestData -> Gen [Mutez]
+genManyBids testData = do
+  len <- Gen.int (Range.linear 0 5)
+  iterateM len (genBid testData) (testOpeningPrice testData)
+
+-- | Generate a valid bid, such that it is:
+--
+-- * Greater than the previous bid or opening price.
+-- * Above either the `min_raise` or the `min_raise_percent` thresholds.
+genBid :: TestData -> Mutez -> Gen Mutez
+genBid testData@TestData{testMinRaise} previousBid = do
+  raiseLowerBound <- Gen.element [testMinRaise, minRaisePercentOfBid testData previousBid]
+  raise <- genMutez' (Range.linear raiseLowerBound (raiseLowerBound + 1000))
+  pure $ previousBid + raise
+
+-- | Given a generator of values of type @a@ and an initial value,
+-- repeatedly uses the generator to create a list of the given length,
+-- feeding it the previously generated value at each iteration.
+iterateM :: forall a. Int -> (a -> Gen a) -> a -> Gen [a]
```

Comment:
This is duplicated between the two modules, but is generic enough to not belong to either, maybe move it to `Test.Util` instead?

Reply:
Done